go/src/lib.rs

```diff
@@ -540,9 +540,12 @@ pub unsafe extern "C" fn command(
         cmd.arg(command_arg);
     }
 
-    let r_bytes = unsafe { std::slice::from_raw_parts(route_bytes, route_bytes_len) };
-
-    let route = Routes::parse_from_bytes(r_bytes).unwrap();
+    let route = if !route_bytes.is_null() {
```

Comment:
We really should have safety documentation for this and remove the TODO. Technically, we could've just stated in our safety documentation that `route_bytes` cannot be null. I think this change is fine, but we need to be clear about what the caller is and is not responsible for.

Reply:
I think we should allow null values here, so that the client wrapper doesn't need to do empty allocations for those pointers. The safety documentation has been updated. While updating the document, I also found `arg_vec` suffered the same unsafe precondition violation issue; therefore, I updated it as well.

Reply:
It is ok to have it null, if it is properly reflected in docs. Thank you for updating docs!